lapack/unit_test/Test_Lapack_svd.hpp

```diff
@@ -0,0 +1,321 @@
+  // are known: sqrt(45) and sqrt(5)
+  EXPECT_NEAR_KK_REL(S_h(0), static_cast<mag_type>(Kokkos::sqrt(45)),
+                     100 * eps);
+  EXPECT_NEAR_KK_REL(S_h(1), static_cast<mag_type>(Kokkos::sqrt(5)), 100 * eps);
+
+  // The singular vectors should be identical
+  // or of oposite sign we check the first
+  // component of the vectors to determine
+  // the proper signed comparison.
+  std::vector<scalar_type> Uref = {
+      static_cast<scalar_type>(1 / Kokkos::sqrt(10)),
+      static_cast<scalar_type>(3 / Kokkos::sqrt(10)),
+      static_cast<scalar_type>(-3 / Kokkos::sqrt(10)),
+      static_cast<scalar_type>(1 / Kokkos::sqrt(10))};
+  std::vector<scalar_type> Vtref = {
+      static_cast<scalar_type>(1 / Kokkos::sqrt(2)),
+      static_cast<scalar_type>(1 / Kokkos::sqrt(2)),
+      static_cast<scalar_type>(-1 / Kokkos::sqrt(2)),
+      static_cast<scalar_type>(1 / Kokkos::sqrt(2))};
+
+  // Both rotations and reflections are valid
+  // vector basis so we need to check both signs
+  // to confirm proper SVD was achieved.
+  if constexpr (KAT_S::is_complex) {
+    mag_type sign_vec =
+        (U_h(0, 0).real() > KAT_S::zero().real()) ? mag_one : -mag_one;
+    EXPECT_NEAR_KK_REL(sign_vec * U_h(0, 0), Uref[0], 100 * eps);
+    EXPECT_NEAR_KK_REL(sign_vec * U_h(1, 0), Uref[1], 100 * eps);
+    sign_vec = (U_h(0, 1).real() < KAT_S::zero().real()) ? mag_one : -mag_one;
+    EXPECT_NEAR_KK_REL(sign_vec * U_h(0, 1), Uref[2], 100 * eps);
+    EXPECT_NEAR_KK_REL(sign_vec * U_h(1, 1), Uref[3], 100 * eps);
+
+    sign_vec = (Vt_h(0, 0).real() > KAT_S::zero().real()) ? mag_one : -mag_one;
+    EXPECT_NEAR_KK_REL(sign_vec * Vt_h(0, 0), Vtref[0], 100 * eps);
+    EXPECT_NEAR_KK_REL(sign_vec * Vt_h(1, 0), Vtref[1], 100 * eps);
+    sign_vec = (Vt_h(0, 1).real() < KAT_S::zero().real()) ? mag_one : -mag_one;
+    EXPECT_NEAR_KK_REL(sign_vec * Vt_h(0, 1), Vtref[2], 100 * eps);
+    EXPECT_NEAR_KK_REL(sign_vec * Vt_h(1, 1), Vtref[3], 100 * eps);
+  } else {
+    mag_type sign_vec = (U_h(0, 0) > KAT_S::zero()) ? mag_one : -mag_one;
+    EXPECT_NEAR_KK_REL(sign_vec * U_h(0, 0), Uref[0], 100 * eps);
+    EXPECT_NEAR_KK_REL(sign_vec * U_h(1, 0), Uref[1], 100 * eps);
+    sign_vec = (U_h(0, 1) < KAT_S::zero()) ? mag_one : -mag_one;
+    EXPECT_NEAR_KK_REL(sign_vec * U_h(0, 1), Uref[2], 100 * eps);
+    EXPECT_NEAR_KK_REL(sign_vec * U_h(1, 1), Uref[3], 100 * eps);
+
+    sign_vec = (Vt_h(0, 0) > KAT_S::zero()) ? mag_one : -mag_one;
+    EXPECT_NEAR_KK_REL(sign_vec * Vt_h(0, 0), Vtref[0], 100 * eps);
+    EXPECT_NEAR_KK_REL(sign_vec * Vt_h(1, 0), Vtref[1], 100 * eps);
+    sign_vec = (Vt_h(0, 1) < KAT_S::zero()) ? mag_one : -mag_one;
+    EXPECT_NEAR_KK_REL(sign_vec * Vt_h(0, 1), Vtref[2], 100 * eps);
+    EXPECT_NEAR_KK_REL(sign_vec * Vt_h(1, 1), Vtref[3], 100 * eps);
+  }
+
+  return 0;
+}
+
+template <class AMatrix, class Device>
+int impl_test_svd(const int m, const int n) {
+  AMatrix A("A", m, n);
+
+  return 0;
```

Comment:
Is this a placeholder for testing the fallback implementation of svd?

Reply:
Actually this will be used to implement a test based on randomly generate matrices of various sizes. I wanted to start with analytical tests to check that the values computed are correct.